db/migrations/sdp-migrations/2025-05-18.1-create-embedded-wallet.sql

```diff
@@ -0,0 +1,28 @@
+-- +migrate Up
+
+CREATE TYPE embedded_wallet_status AS ENUM(
+    'PENDING',
+    'PROCESSING',
+    'FAILED',
+    'SUCCESS'
+);
+
+CREATE TABLE embedded_wallets (
+    token VARCHAR(36) PRIMARY KEY DEFAULT public.uuid_generate_v4(),
+    tenant_id VARCHAR(36) NOT NULL,
+    wasm_hash VARCHAR(64),
+    contract_address VARCHAR(56),
+    created_at TIMESTAMP WITH TIME ZONE NOT NULL DEFAULT NOW(),
+    updated_at TIMESTAMP WITH TIME ZONE NOT NULL DEFAULT NOW(),
+    wallet_status embedded_wallet_status NOT NULL DEFAULT 'PENDING'::embedded_wallet_status
+);
+
+CREATE TRIGGER refresh_wallets_updated_at BEFORE UPDATE ON embedded_wallets FOR EACH ROW EXECUTE PROCEDURE update_at_refresh();
```

Comment:
I suggest renaming the trigger to make it clear it relates to the `embedded_wallets` table, and not the `wallets` table:

```suggestion
CREATE TRIGGER refresh_embedded_wallets_updated_at BEFORE UPDATE ON embedded_wallets FOR EACH ROW EXECUTE PROCEDURE update_at_refresh();
```